### ISSUE TITLE
Fix S3875 FP: Do not raise an issue when implementing IEqualityOperators interface

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#6312]
### Description

FP for [S3875](https://sonarsource.github.io/rspec/#/rspec/S3875/csharp) (DotNotOverloadOperatorEqual).

When implementing IEqualityOperators  interface the rule should not raise.

### Repro steps

```
    class MyClass : IEqualityOperators<MyClass, MyClass, MyClass>
    {
        public static MyClass operator ==(MyClass? left, MyClass? right) => new MyClass(); // Noncompliant FP (implementing IEqualityOperators interface require == operator overload)
        
        public static MyClass operator !=(MyClass? left, MyClass? right) => new MyClass();
    }
```

### Expected behavior

The issue should not be raised in this case.

### Actual behavior

An issue is raised.
